openquake/hazardlib/gsim/mgmpe/modifiable_gmpe.py

```diff
@@ -22,13 +22,104 @@
 import numpy as np
 from openquake.hazardlib.gsim.base import GMPE, registry, CoeffsTable
 from openquake.hazardlib.contexts import STD_TYPES
-from openquake.hazardlib.imt import from_string
 from openquake.hazardlib.const import StdDev
+from openquake.hazardlib import const, contexts
+from openquake.hazardlib.imt import from_string, PGA, PGV
+from openquake.hazardlib.const import IMC
+
 
 IMT_DEPENDENT_KEYS = ["set_scale_median_vector",
                       "set_scale_total_sigma_vector",
                       "set_fixed_total_sigma"]
 
+COEFF = {IMC.AVERAGE_HORIZONTAL: [1, 1, 0.01, 0.02, 1],
+         IMC.GMRotI50: [1, 1, 0.03, 0.04, 1],
+         IMC.RANDOM_HORIZONTAL: [1, 1, 0.07, 0.11, 1.05],
+         IMC.GREATER_OF_TWO_HORIZONTAL:
+         [0.1, 1.117, 0.53, 1.165, 4.48, 1.195, 8.70, 1.266, 1.266],
+         IMC.RotD50:
+         [0.09, 1.009, 0.58, 1.028, 4.59, 1.042, 8.93, 1.077, 1.077]}
+
+COEFF_PGA_PGV = {IMC.AVERAGE_HORIZONTAL: [1, 0.01, 1, 1, 0.01, 1],
+                 IMC.GMRotI50: [1, 0.02, 1, 1, 0.03, 1],
+                 IMC.RANDOM_HORIZONTAL: [1, 0.07, 1.03],
+                 IMC.GREATER_OF_TWO_HORIZONTAL: [1.117, 0, 1, 1, 0, 1],
+                 IMC.RotD50: [1.009, 0, 1, 1, 0, 1]}
+
+
+def horiz_comp_to_geom_mean(self, ctx, imt):
+    """
+    This function converts ground-motion obtained for a given description of
+    horizontal component into ground-motion values for geometric_mean.
+    The conversion equations used are from:
+        - Beyer and Bommer (2006): for arithmetic mean, GMRot and random
+        - Boore and Kishida (2017): for RotD50
+    """
+
+    # Get the definition of the horizontal component using in the original GMM
+    horcom = getattr(self.gmpe, 'DEFINED_FOR_INTENSITY_MEASURE_COMPONENT')
+
+    # IMT period
+    T = imt.period
+
+    # Conversion coefficients
+    C = COEFF[horcom]
+    C_PGA_PGV = COEFF_PGA_PGV[horcom]
+
+    # Get the string defining the horizontal component
+    comp = str(horcom).split('.')[1]
+
+    # List of the horizontal component definitions that can be converted into
+    # geometric mean
+    tmp = ['AVERAGE_HORIZONTAL', 'GMRotI50', 'RANDOM_HORIZONTAL',
+           'GREATER_OF_TWO_HORIZONTAL', 'RotD50']
+
+    # Apply the conversion
+    if comp in tmp:
+        if imt == PGA():
```

Comment:
Better to use `if imt.name == 'PGA'` or `imt.name == 'PGV'` to avoid a useless instantiation.